src/main/java/xyz/nucleoid/plasmid/util/BucketFind.java

```diff
@@ -0,0 +1,44 @@
+package xyz.nucleoid.plasmid.util;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Predicate;
+
+public final class BucketFind {
+    /**
+     * Finds any directly connected blocks and puts them in a set.
+     *
+     * @param origin The position of the first block
+     * @param predicate The predicate for the blocks that can be accepted in the set
+     * @param limit The amount of maximum blocks to find
+     */
+    public static Set<BlockPos> find(ServerWorld world, BlockPos origin, Predicate<BlockState> predicate, int limit) {
+        Set<BlockPos> set = new HashSet<>();
+        Set<BlockPos> ends = new HashSet<>();
+        ends.add(origin);
+        while(limit > 0) {
+            if(ends.isEmpty()) {
+                return set;
+            }
+            BlockPos pos = ends.stream().findAny().get();
+            for(Direction direction : Direction.values()) {
+                BlockPos.Mutable local = pos.offset(direction).mutableCopy();
```

Comment:
Extract the `BlockPos.Mutable` from the loop, and reuse a single instance.

Reply:
Is it better in e00ec75575a0bbea010b766f1e21dbdaf7f06de7?

Reply:
No, because you are now just assigning multiple `BlockPos.Mutable` instances to the same local variable.

Reply:
Forgot about `BlockPos.Mutable#move` 😅 is it good in 6d9e3f3?